### PR TITLE
Add CSS `display-none` utility

### LINKED ISSUE #773: Add `oui-displayNone` CSS utility
**Is your feature request related to a problem? Please describe.**

Currently, there is no way to apply `display: none` in OUI. This is important for scenarios when we need access to the DOM elements without actually showing them. For example:

```tsx
const ref = useRef(null);
const [show, setShow] = useState(false);

useEffect(() => {
  // populate ref with DOM elements after some lengthy operation

  setShow(true);
}, [ref]);

return (
  <div ref={ref} style={{ display: show ? 'block' : 'none' }} />
);
```

**Describe the solution you'd like**

The simplest and cleanest way to achieve this would be to add a new CSS utility class called `oui-displayNone`. This would be the same as the other display utilities, but with `display` set to `none`.

Other display utilities:

https://github.com/opensearch-project/oui/blob/71dc4bd5f04603d69143a94c528fb57bc960c7ce/src/global_styling/utility/_utility.scss#L21-L24

Proposed new utility:

```scss
.oui-displayNone {display: none !important;}
```

**Describe alternatives you've considered**

I also considered creating a utility component to achieve this. It might look something like this:

```tsx
return (
  <OuiConditionalRender condition={/* boolean value */}>
    {/* content that is display: none when condition is false */}
  </OuiConditionalRender>
);
```

However, given the simplicity of the need, I think having an entire component for this would be extremely overkill and not worth the maintaining effort.

**Additional context**

This gap was identified in an OUI compliance audit: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4122

### PR DESCRIPTION
**Pull request by ShatilKhan**
### Description

Added new CSS Utility Class `display-none`

### Issues Resolved

Fixes #773 

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] All tests pass
  - [ ] `yarn lint`
  - [ ] `yarn test-unit`
- [X] Update [CHANGELOG.md](./../CHANGELOG.md)
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/oui/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by joshuarrrr — COMMENTED** [review]
@ShatilKhan Can you also update https://github.com/opensearch-project/oui/blob/main/src-docs/src/views/utility_classes/utility_classes.js so that the new class is listed on https://oui.opensearch.org/1.3/#/utilities/css-utility-classes

**Comment by ShatilKhan** [comment]
> @ShatilKhan Can you also update https://github.com/opensearch-project/oui/blob/main/src-docs/src/views/utility_classes/utility_classes.js so that the new class is listed on https://oui.opensearch.org/1.3/#/utilities/css-utility-classes

Yeah I'll update it today

**Comment by joshuarrrr** [comment]
@ShatilKhan Just checking back to see if you'll be able to finish up this PR.

**Comment by ShatilKhan** [comment]
> @ShatilKhan Can you also update https://github.com/opensearch-project/oui/blob/main/src-docs/src/views/utility_classes/utility_classes.js so that the new class is listed on https://oui.opensearch.org/1.3/#/utilities/css-utility-classes

Done with this one

File: src-docs/src/views/utility_classes/utility_classes.js [hunk 1]
```diff
@@ -93,6 +93,24 @@ export default () => (
...
   /* Your content */
+</span>`}
+    />
+        <OuiSpacer />
+    <UtilityClassesSection
+      code="oui-displayNone"
+      description={
+        <p>
+          Changes the element’s display property to{' '}
+          <OuiCode language="sass">display: None;</OuiCode>
```
Comment (joshuarrrr) [inline src-docs/src/views/utility_classes/utility_classes.js:104]:
```suggestion
          <OuiCode language="sass">display: none;</OuiCode> 
```

File: src-docs/src/views/utility_classes/utility_classes.js [hunk 2]
```diff
@@ -93,6 +93,24 @@ export default () => (
...
+      code="oui-displayNone"
+      description={
+        <p>
+          Changes the element’s display property to{' '}
+          <OuiCode language="sass">display: None;</OuiCode> 
+        </p>
+      }
+      example={
+        <OuiMark style={wrappingExampleStyle} className="oui-displayNone">
+          Displaying None
```
Comment (joshuarrrr) [inline src-docs/src/views/utility_classes/utility_classes.js:109]:
```suggestion
          Displaying none
```

File: src-docs/src/views/utility_classes/utility_classes.js [hunk 3]
```diff
@@ -93,6 +93,24 @@ export default () => (
       }
       snippet={`<span className="oui-fullWidth">
   /* Your content */
+</span>`}
+    />
+        <OuiSpacer />
+    <UtilityClassesSection
```
Comment (joshuarrrr) [inline src-docs/src/views/utility_classes/utility_classes.js:99]: indentation

```suggestion
    <OuiSpacer />
    <UtilityClassesSection
```

**Review by joshuarrrr — COMMENTED** [review]
Thanks @ShatilKhan! Just a few minor formatting tweaks.

**Comment by ShatilKhan** [comment]
Done with the suggested errors

**Review by joshuarrrr — DISMISSED** [review]

**Comment by BSFishy** [comment]
This linting error should be fixed: https://github.com/opensearch-project/oui/actions/runs/7175077016/job/19539863193?pr=1120#step:5:17

**Review by ShatilKhan — COMMENTED** [review]
Fixed linting problem on line 104 in utility_classes.js

**Review by BSFishy — COMMENTED** [review]
I just realized this should also be added to `src/themes/oui-next/global_styling/utility/_utility.scss` so it works in next themes as well

**Comment by ShatilKhan** [comment]
> I just realized this should also be added to `src/themes/oui-next/global_styling/utility/_utility.scss` so it works in next themes as well

done

**Review by BSFishy — APPROVED** [review]

**Pull request merged**